recipes/_update_shcluster.rb

```diff
@@ -0,0 +1,31 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Recipe:: _update_shcluster
+#
+# Recipe determines whether to add/remove this SH member to/from SH Cluster.
+
+password_file = File.join node['splunk']['external_config_directory'], 'password'
+search_heads = CernerSplunk.my_cluster_data(node)['sh_cluster']
+
+fail 'Search Heads are not configured for sh clustering in the cluster databag' if search_heads.nil? || search_heads.empty?
+
+unless node['splunk']['config']['sh_member_action'].nil?
+  if node['splunk']['config']['sh_member_action'] == 'add'
+
+    execute "#{node['splunk']['config']['sh_member_action']} the search head" do
+      command "cat #{password_file} | xargs -I{} #{node['splunk']['cmd']} add shcluster-member -current_member_uri #{search_heads.first} -auth admin:{}"
```

Comment:
Instead of propagating knowledge on how we keep track of the admin password around, with the cat -> xargs pattern, should we be pulling the admin password into the node run state? (Then we can use it in lazy attributes), and as long as we update it when we update the admin password we don't care about order as much.